datetime/src/test/java/com/fasterxml/jackson/datatype/jsr310/deser/MonthDeserTest.java

```diff
@@ -0,0 +1,230 @@
+        assertEquals(Month.DECEMBER, value);
+    }
+
+
+    @Test
+    public void _testDeserializationWithTypeInfo01_default() throws Exception
+    {
+        ObjectMapper mapper = new ObjectMapper()
+            .registerModule(new JavaTimeModule());
+        mapper.addMixIn(TemporalAccessor.class, MockObjectConfiguration.class);
+
+        TemporalAccessor value = mapper.readValue("[\"java.time.Month\",\"11\"]", TemporalAccessor.class);
+        assertEquals(Month.DECEMBER, value);
+    }
+
+    @Test
+    public void _testDeserializationWithTypeInfo01_feature() throws Exception
+    {
+        ObjectMapper mapper = new ObjectMapper()
+            .registerModule(new JavaTimeModule().enable(JavaTimeFeature.ONE_BASED_MONTHS));
+        mapper.addMixIn(TemporalAccessor.class, MockObjectConfiguration.class);
+
+        TemporalAccessor value = mapper.readValue("[\"java.time.Month\",\"11\"]", TemporalAccessor.class);
+        assertEquals(Month.NOVEMBER, value);
+    }
+
+
+
+    static class Wrapper {
+        @JsonFormat(pattern="MM")
+        public Month value;
+
+        public Wrapper(Month v) { value = v; }
+    }
+
+    @Test
+    public void testFormatAnnotation() throws Exception
+    {
+        ObjectMapper MAPPER = newMapper();
+        String json = newMapper().writeValueAsString(new Wrapper(Month.DECEMBER));
+        assertEquals("{\"value\":\"12\"}", json);
+
+        Wrapper output = MAPPER.readValue(json, Wrapper.class);
+        assertEquals(new Wrapper(Month.of(12)).value, output.value);
+    }
+
+    /*
+    /**********************************************************
+    /* Tests for empty string handling
+    /**********************************************************
+     */
+
+    // minor changes in 2.12
+    @Test
+    public void testDeserializeFromEmptyString() throws Exception
+    {
+        // First: by default, lenient, so empty String fine
+        TypeReference<Map<String, Month>> MAP_TYPE_REF = new TypeReference<Map<String, Month>>() { };
+        ObjectReader objectReader = newMapper().registerModule(new JavaTimeModule())
+            .readerFor(MAP_TYPE_REF);
+
+        Map<String, Month> map = objectReader.readValue("{\"month\":null}");
+        assertNull(map.get("month"));
+
+        Map<String, Month> map2 = objectReader.readValue("{\"month\":\"\"}");
+        assertNotNull(map2);
+
+        // But can make strict:
+        ObjectMapper strictMapper = mapperBuilder()
+            .addModule(new JavaTimeModule())
+            .build();
+        strictMapper.configOverride(Month.class)
+            .setFormat(JsonFormat.Value.forLeniency(false));
+
+        try {
+            strictMapper.readerFor(MAP_TYPE_REF).readValue("{\"date\":\"\"}");
+            fail("Should not pass");
+        } catch (MismatchedInputException e) {
+            verifyException(e, "not allowed because 'strict' mode set for");
+        }
+    }
+
+    private void expectFailure(String aposJson) throws Throwable {
```

Comment:
I know existing tests use this helper method, but it's.... pretty bad. I wish I had time to fix all of these from the original implementation that was contributed.

But basically verification should check that:

1. Exception class is EXACTLY what we expect and
2. Exception message matches completely, or relevant part does

so this helper method can be avoided and verification inlined in test methods.